lib/modules/dosomething/dosomething_reportback/dosomething_reportback.query.inc

```diff
@@ -143,3 +143,26 @@ function dosomething_reportback_filter_flagged_reportbacks($params) {
   return $params;
 }
 
+/**
+ * Filter out requested Reportbacks based on associated Signup. 
+ *
+ * @param string $run_id Campaign run id. 
+ * @param string $sid Signup id. 
+ * @return mixed
+ */
+function dosomething_reportback_filter_reportbacks_by_signups($run_id, $sid) {
+  $reportbacks = db_query("SELECT DISTINCT s.sid, r.rbid, r.run_nid, s.run_nid
+                          FROM dosomething_signup s 
+                          INNER JOIN dosomething_reportback r on s.run_nid = r.run_nid
+                          WHERE s.uid = r.uid;");
```

Comment:
Also, `dosomething_reportback_filter_reportbacks_by_signups` does some logic to get rid of the reportbacks with `reportback_run` of `0` and return a user's reportbacks by signup id. 

I'm not sure where to put that logic - it seems like it'd be too much going on if we placed after `$result = $query->execute();` on line 28 in `dosomething_signup.query.inc`?